spec/odp/odp_event_manager_spec.rb

```diff
@@ -385,6 +385,22 @@
       event_manager.stop!
     end
 
+    it 'should flush when flush interval is zero' do
+      allow(SecureRandom).to receive(:uuid).and_return(test_uuid)
+      event_manager = Optimizely::OdpEventManager.new(logger: spy_logger)
+      allow(event_manager.api_manager).to receive(:send_odp_events).once.with(api_key, api_host, odp_events).and_return(false)
+      event_manager.instance_variable_set('@flush_interval', 0.0)
+      event_manager.start!(odp_config)
+
+      event_manager.send_event(**events[0])
+      event_manager.send_event(**events[1])
+      sleep(0.1) until event_manager.instance_variable_get('@event_queue').empty?
```

Comment:
Interesting here, that you added a tiny sleep time. In py-sdk we worked hard to avoid it :-p. 
Ruby threading is different?

Reply:
Yea it's not ideal 😄. Here we are looping, checking to see if the queue is empty. This is required because ruby's SizedQueue doesn't have a wait/join method.